### DISCUSSION TITLE
[TPA] Error spark.http.matching.GeneralError luego de deployar

### DISCUSSION THREAD

**Discussion by gastonbaeza** [dds-jv/foro#156] [Consultas de herramientas] [answered]
Buenas, 
Deployé la app en Heroku, puedo ver el formulario de login en la web. 
El problema ocurre al intentar loguear un usuario, dado que las tablas nunca se crearon en la base de datos. (Esto lo pude ver ingresando a la base con HeidiSQL).

La configuración del persistence.xml es la siguiente:
```
<property name="hibernate.archive.autodetection" value="class"/>
<property name="hibernate.connection.driver_class" value="com.mysql.jdbc.Driver"/>
<property name="hibernate.connection.url" value="jdbc:mysql://mysql://sm6bvb1r4mx7fdmp:fkkqt35dc3ek0x0u@de1tmi3t63foh7fa.cbetxkdyhwsb.us-east-1.rds.amazonaws.com:3306/base"/>
<property name="hibernate.connection.username" value="username"/>
<property name="hibernate.connection.password" value="password"/>
<property name="hibernate.dialect" value="org.hibernate.dialect.MySQL5Dialect"/>
/*Censuro los valores de base, username y password*/

<property name="hibernate.show_sql" value="true"/>
<property name="hibernate.format_sql" value="true"/>
<property name="use_sql_comments" value="true"/>
<property name="hibernate.hbm2ddl.auto" value="create"/>
```

Y la excepción que puedo ver en los logs de Heroku:
```
ERROR spark.http.matching.GeneralError  -  java.lang.NullPointerException
	at org.uqbarproject.jpa.java8.extras.PerThreadEntityManagers.getEntityManager(PerThreadEntityManagers.java:27)
	at org.uqbarproject.jpa.java8.extras.WithGlobalEntityManager.entityManager(WithGlobalEntityManager.java:8)
 	at repositorios.RepositorioUsuarios.obtenerUsuarioPorCredenciales(RepositorioUsuarios.java:29)
       ...
```

Demás está aclarar que la aplicación funcionaba antes de los cambios necesarios para deployar en Heroku y los hice siguiendo la guía recomendada (https://github.com/dds-jv/foro/issues/87#issuecomment-736948631).

Por el log, realmente no me queda claro si el error se relaciona con Spark o con el EntityManager.
¿Cómo puedo comprobar que la conexión a la base se realiza correctamente?

**Comment by gastonprieto** [comment]
Buenas @gastonbaeza 

Habría que ver el log completo, lo que aparece ahí es el nullpointer por no conseguir el `entityManager`. Me suena que tenes algún dato de conexión que esta incorrecto.

**Comment by gastonbaeza** [comment, marked as answer]
Se solucionó agregando la dependencia:

`<dependency>
    <groupId>org.javassist</groupId>
    <artifactId>javassist</artifactId>
    <version>3.22.0-GA</version>
</dependency>`

Fuente: https://stackoverflow.com/questions/27202939/could-not-build-classfile-archiveexception/57191682#57191682
